### ISSUE TITLE
Async operations on cookie jar

### ISSUE THREAD

**Issue by fredrikekelund** [cgiffard/node-simplecrawler#313]
I haven't given much thought to this before now, but for the crawler to be able to run reliably in a distributed fashion, we need to have a distributed cookie jar as well. To have that, we need to only use asynchronous operations on the cookie jar.

I'd like to hold off till after 1.0.0 with this, though. Changing the crawler's internal behaviour doesn't break anything for the end user. And the cookie API is quite poorly documented, so I think if we add a deprecation warning first about the fact that the cookie jar methods will stop returning results (and instead only use callbacks) we should be fine

**Comment by cgiffard** [comment]
Yeah, let's call it a 2.0 release (because it's breaking, not because it's far away.) But I don't think it's so risky — from what I've seen most people use the cookie jar implicitly rather than hook any of their own stuff up to it.

**Comment by fredrikekelund** [comment]
I think we should look into whether using a 3rd party dependency for cookie management makes sense or not.

[tough-cookie](https://www.npmjs.com/package/tough-cookie) is widely used (being request's cookie module), and comes with a cookie store which uses a replaceable, asynchronous back-end. There are even back-end implementations for both redis and SQL-databases.
